src/ape_ethereum/ecosystem.py

```diff
@@ -317,6 +318,33 @@ class Block(BlockAPI):
     def validate_ints(cls, value):
         return to_int(value) if value else 0
 
+    @computed_field()  # type: ignore[misc]
+    @property
+    def size(self) -> int:
+        """
+        The size of the block in gas. Most of the time,
+        this field is passed to the model at validation time,
+        but occasionally it is missing (like in Infura::newHeads),
+        in which case it gets calculated.
+        """
+
+        if self._size is not None:
+            # The size was provided with the rest of the model
+            # (normal).
+            return self._size
+
+        # Try to get it from the provider.
+        if provider := self.network_manager.active_provider:
+            block = provider.get_block(self.number)
+            size = block._size
+            if size is not None and size > -1:
+                self._size = size
+                return size
+
+        # Don't cache.
+        # TODO: Could try to manually calculate here.
+        return -1
```

Comment:
Same as prior note, raise `NotImplementedError` to prevent serialization/caching?